dotnet/src/SemanticKernel.Skills/Skills.OpenAPI/Model/RestApiOperation.cs

```diff
@@ -101,6 +104,62 @@ public Uri BuildOperationUrl(IDictionary<string, string> arguments)
         return new Uri($"{serverUrl.TrimEnd('/')}/{path.TrimStart('/')}", UriKind.Absolute);
     }
 
+    /// <summary>
+    /// Renders operation request headers.
+    /// </summary>
+    /// <param name="arguments">The operation arguments.</param>
+    /// <returns>The rendered request headers.</returns>
+    public IDictionary<string, string> RenderHeaders(IDictionary<string, string> arguments)
+    {
+        var headersMetadata = this.Parameters.Where(p => p.Location == RestApiOperationParameterLocation.Header);
+
+        var headers = new Dictionary<string, string>();
+
+        foreach (var header in this.Headers)
+        {
+            var headerName = header.Key;
+            var headerValue = header.Value;
+
+            //A try to resolve header value in arguments.
+            if (arguments.TryGetValue(headerName, out var value))
+            {
+                headers.Add(headerName, value);
+                continue;
+            }
+
+            //Header value is already supplied.
+            if (!string.IsNullOrEmpty(headerValue))
+            {
+                headers.Add(headerName, headerValue);
+                continue;
+            }
+
+            //Getting metadata for the header
+            var headerMetadata = headersMetadata.FirstOrDefault(hm => hm.Name == headerName);
+            if (headerMetadata == null)//No metadata found for the header.
```

Comment:
[nit] Put this comment above or inside the if block: 
```
if (headerMetadata == null)
{
    //No metadata found for the header.
    throw new RestApiOperationException($"No value for the '{headerName} header is found.'");
}
```

Reply:
Looks a little bit better. :)